tests/platform_tests/test_reboot.py

```diff
@@ -277,3 +277,21 @@ def test_continuous_reboot(duthosts, enum_rand_one_per_hwsku_hostname,
     pytest_assert(ls_ending_out == ls_starting_out,
                   "Console devices have changed: expected console devices: {}, got: {}"
                   .format(", ".join(sorted(ls_starting_out)), ", ".join(sorted(ls_ending_out))))
+
+
+def test_fsck_after_reboot(duthosts, enum_rand_one_per_hwsku_hostname,
+                           localhost, conn_graph_facts, xcvr_skip_list):        # noqa F811
+    """
+    @summary: This test case is checking syslog to verify fsck script has run for
+    checking and repairing the file system when boot in initramfs stage.
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+
+    # Call reboot to run fsck for repair file system.
+    reboot_and_check(localhost, duthost, conn_graph_facts["device_conn"][duthost.hostname],
+                     xcvr_skip_list, reboot_type=REBOOT_TYPE_COLD, duthosts=duthosts)
+
+    # Check fsck ran by syslog entry.
+    cmd = "sudo find /var/log/syslog* -type f -mmin -5 -exec zgrep 'fsck' {} +"
```

Comment:
@xincunli-sonic `find /var/log/syslog*` this will include past syslog files not related to last reboot so how can one be sure the grep is actually for last reboot?

Reply:
Yes,  you are right, find with -mmin -5 will only grab the fild has been modified in last 5 minutes instead of log entry time stamp. Change to `journalctl` for last 5 minutes log entried which contains the keyword.